File: java/modelo/PersonalSaludFactory.java
package modelo;
import java.time.LocalDate;
public class PersonalSaludFactory {
    public static PersonalSalud crearProfesional(String rol, String dni, String nombres, 
                                                 String apellidos, LocalDate fechaNac, 
                                                 String sexo, String colegiatura, String parametroExtra) {
        if (rol == null) {
            throw new IllegalArgumentException("El rol profesional no puede ser nulo.");
        }
        switch (rol.toUpperCase().trim()) {
            case "MEDICO":
                return new Medico(dni, nombres, apellidos, fechaNac, sexo, colegiatura, parametroExtra);
            case "ENFERMERO":
                String programa = (parametroExtra == null || parametroExtra.trim().isEmpty()) ? "CRED" : parametroExtra;
                return new Enfermero(dni, nombres, apellidos, fechaNac, sexo, colegiatura, programa);
            default:
                throw new IllegalArgumentException("Rol profesional no soportado: " + rol);
        }
    }
}

File: python/modelo/personal_salud_factory.py
from datetime import date
from modelo.medico import Medico
from modelo.enfermero import Enfermero
from modelo.personal_salud import PersonalSalud

class PersonalSaludFactory:
    @staticmethod
    def crear_profesional(rol: str, dni: str, nombres: str, apellidos: str, 
                          fecha_nac: date, sexo: str, colegiatura: str, parametro_extra: str = "") -> PersonalSalud:
        rol_limpio = rol.strip().upper() if rol else ""
        if rol_limpio == "MEDICO":
            return Medico(dni, nombres, apellidos, fecha_nac, sexo, colegiatura, rne=parametro_extra)
        elif rol_limpio == "ENFERMERO":
            prog = parametro_extra if parametro_extra else "CRED"
            return Enfermero(dni, nombres, apellidos, fecha_nac, sexo, colegiatura, programa=prog)
        else:
            raise ValueError(f"Rol profesional no reconocido: {rol}")